Neos.ContentRepository.Core/Classes/CommandHandler/CommandBus.php

```diff
@@ -4,38 +4,38 @@
 
 namespace Neos\ContentRepository\Core\CommandHandler;
 
-use Neos\ContentRepository\Core\CommandHandlingDependencies;
 use Neos\ContentRepository\Core\ContentRepository;
 use Neos\ContentRepository\Core\EventStore\EventsToPublish;
-use Neos\ContentRepository\Core\EventStore\EventsToPublishFailed;
 
 /**
  * Implementation Detail of {@see ContentRepository::handle}, which does the command dispatching to the different
  * {@see CommandHandlerInterface} implementation.
  *
  * @internal
  */
-final class CommandBus
+final readonly class CommandBus
 {
     /**
      * @var CommandHandlerInterface[]
      */
-    private array $handlers;
+    public array $handlers;
```

Comment:
Not a blocker but I would prefer the handlers not to be publicly exposed here.
AFAIS this is only used in the CR factory:
```php
$publicCommandBus = new CommandBus(
    $commandHandlingDependencies,
    new ContentStreamCommandHandler(),
    new WorkspaceCommandHandler(
        $commandSimulatorFactory,
        $this->projectionFactoryDependencies->eventStore,
        $this->projectionFactoryDependencies->eventNormalizer,
    ),
    ...$commandBusForRebaseAbleCommands->handlers
);
```
but maybe we can do something like
```php
$publicCommandBus = $commandBusForRebaseAbleCommands->withAdditionalHandlers(
  $commandHandlingDependencies,
  new ContentStreamCommandHandler(),
  new WorkspaceCommandHandler(
      $commandSimulatorFactory,
      $this->projectionFactoryDependencies->eventStore,
      $this->projectionFactoryDependencies->eventNormalizer,
  ),
);
```

instead to keep it private?
(I know the CommandBus is `@internal` but...?

Reply:
hmm i agree... we only need it for the factory ... 3rd option store them in a array and use that ... ill do that:)

Reply:
nevermind `withAdditionalHandlers ` works better :D